daemon/set.go

```diff
@@ -0,0 +1,53 @@
+package daemon
+
+import (
+	"github.com/docker/docker/runconfig"
+)
+
+func (daemon *Daemon) ContainerSet(name string, hostConfig *runconfig.HostConfig) ([]string, error) {
+	warnings, err := daemon.verifyHostConfig(hostConfig)
+	if err != nil {
+		return warnings, err
+	}
+
+	container, err := daemon.Get(name)
+	if err != nil {
+		return warnings, err
+	}
+
+	daemon.updateResources(container, hostConfig)
+
+	if err := daemon.Set(container.command); err != nil {
+		return warnings, err
+	}
+
+	return warnings, nil
+}
+
+func (daemon *Daemon) updateResources(container *Container, hostConfig *runconfig.HostConfig) {
```

Comment:
I would kind of prefer if the logic of updating the container properties were done by the container itself and not in the daemon.  IOW, have the daemon.updateResorces() & daemon.Set() calls above be a container.Set() (or 'Update') call instead.  Then we can add logging to that func too.